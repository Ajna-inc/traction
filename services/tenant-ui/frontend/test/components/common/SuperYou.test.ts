import { mount } from '@vue/test-utils';
import PrimeVue from 'primevue/config';
import { describe, expect, test } from 'vitest';

import SuperYou from '@/components/common/SuperYou.vue';

describe('SuperYou', () => {
  test('mount matches snapshot with expected values', async () => {
    const wrapper = mount(SuperYou, {
      props: {
        apiUrl: 'http://test.com',
        templateJson: {
          name: 'test',
        },
        icon: 'pi-user',
        text: 'test test',
      },
      global: {
        plugins: [PrimeVue],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
