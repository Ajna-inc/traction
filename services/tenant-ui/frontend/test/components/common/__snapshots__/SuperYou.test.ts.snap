// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SuperYou > mount matches snapshot with expected values 1`] = `
"<button class=\\"p-button p-component\\" type=\\"button\\" aria-label=\\"test test\\" title=\\"Custom API Call\\"><span class=\\"p-button-icon p-button-icon-left pi pi-user\\"></span><span class=\\"p-button-label\\">test test</span>
  <!---->
</button>
<!---->"
`;
